core/node/state_keeper/src/updates/l2_block_updates.rs

```diff
@@ -1,23 +1,46 @@
 use std::collections::HashMap;
 
+use once_cell::sync::Lazy;
 use zksync_multivm::{
     interface::{
         CompressedBytecodeInfo, ExecutionResult, L2BlockEnv, TransactionExecutionResult,
-        TxExecutionStatus, VmExecutionMetrics, VmExecutionResultAndLogs,
+        TxExecutionStatus, VmEvent, VmExecutionMetrics, VmExecutionResultAndLogs,
     },
     vm_latest::TransactionVmExt,
 };
+use zksync_system_constants::KNOWN_CODES_STORAGE_ADDRESS;
 use zksync_types::{
     block::{BlockGasCount, L2BlockHasher},
-    event::extract_bytecodes_marked_as_known,
+    ethabi,
     l2_to_l1_log::{SystemL2ToL1Log, UserL2ToL1Log},
     vm_trace::Call,
-    L2BlockNumber, ProtocolVersionId, StorageLogWithPreviousValue, Transaction, VmEvent, H256,
+    L2BlockNumber, ProtocolVersionId, StorageLogWithPreviousValue, Transaction, H256,
 };
 use zksync_utils::bytecode::hash_bytecode;
 
 use crate::metrics::KEEPER_METRICS;
 
+/// Extracts all bytecodes marked as known on the system contracts.
+fn extract_bytecodes_marked_as_known(all_generated_events: &[VmEvent]) -> Vec<H256> {
```

Comment:
Why isn't it a part of `VmEvent` impl?

Reply:
In general, I've moved all `VmEvent` uses as close to their uses as possible; thus, all single-case uses were moved to the higher-level crates. I'm not sure this is the best approach, so if you say that this functionality can actually be reused in the future, I'll move it to `VmEvent`.

Reply:
I see, no, I don't think it will be reused in future. Makes sense to leave it here